lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +18,66 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
+    require 'win32/registry'
+
+    # Ensure the symlink folder exists and is pointing to the proper location.
+    # This also ensures the directory structure is in place prior to creating
+    # the symlink, otherwise the symlink creation fails.
+    def set_symlink_folder
+      puppet_modules_vendor_folder = "#{get_program_files_dir}\\WindowsPowerShell\\Modules\\PuppetVendoredModules"
+      recreate_link = false
+
+      if Puppet::FileSystem.exist?(puppet_modules_vendor_folder)
+        if Puppet::FileSystem.symlink?(puppet_modules_vendor_folder)
+          return if Puppet::FileSystem.readlink(puppet_modules_vendor_folder) == vendored_modules_path
```

Comment:
Jumping out early here. Trying to reduce the additional symlink call if we've determined we are all good. This will be the normal code path 99% of the time so I tried to optimize a bit for it.